### PR TITLE
detect if the disk cannot keep up with data collection

### LINKED ISSUE #7043: Generate alarms if the disk cannot keep up with data collection
##### Feature idea summary
If the disk cannot keep up with metric collection this condition should be detectable and alarms should be generated. This is for `memory mode = dbengine`.

### PR DESCRIPTION
**Pull request by mfundul**
##### Summary
Fixes #7043 
##### Component Name
database/engine

### CODE REVIEW

File: health/health.d/dbengine.conf [hunk 1]
```diff
@@ -23,4 +23,16 @@ lookup: sum -10m unaligned of I/O errors
...
+    on: netdata.dbengine_global_errors
+    os: linux freebsd macos
+ hosts: *
+lookup: sum -10m unaligned of flushing errors
+ units: errors
+ every: 3s
+  crit: $this > 0
+ delay: down 1h multiplier 1.5 max 3h
+  info: number of times dbengine failed to flush in time the last 10 minutes, metric data will not be stored in the database, please reduce disk load or use a faster disk
     to: sysadmin
```
Comment (thiagoftsm) [inline health/health.d/dbengine.conf]: Please add a new line at the end of this file.

File: configs.signatures [hunk 2]
```diff
@@ -381,7 +381,7 @@ declare -A configs_signatures=(
   ['7deb236ec68a512b9bdd18e6a51d76f7']='python.d/mysql.conf'
   ['7e5fc1644aa7a54f9dbb1bd102521b09']='health.d/memcached.conf'
   ['7f13631183fbdf79c21c8e5a171e9b34']='health.d/zfs.conf'
-  ['e48b89d4a97b96acf9a88970ab858c3b']='health.d/dbengine.conf'
+  ['f9ea512f96fc3971e1314d790e9a96ea']='health.d/dbengine.conf'
```
Comment (ilyam8) [inline configs.signatures]: We shouldnt change the signatures by hand.

Comment (mfundul) [reply configs.signatures]: Is there a way to do it automatically? :D

Comment (ilyam8) [reply configs.signatures]: Well. If i remember it correctly this file is not used at all. We always rewrite stock config, we never rewrite user config. These signatures were used before stock/user configs separation.

See #6989. I updated `mysql.conf` there and i changed no singnatures.

Comment (ilyam8) [reply configs.signatures]: @knatsakis correct me i i am wrong

File: health/health.d/dbengine.conf [hunk 3]
```diff
@@ -23,4 +23,16 @@ lookup: sum -10m unaligned of I/O errors
...
+ alarm: 10min_dbengine_global_flushing_errors
+    on: netdata.dbengine_global_errors
+    os: linux freebsd macos
+ hosts: *
+lookup: sum -10m unaligned of flushing errors
+ units: errors
+ every: 3s
+  crit: $this > 0
+ delay: down 1h multiplier 1.5 max 3h
+  info: number of times dbengine failed to flush in time the last 10 minutes, metric data will not be stored in the database, please reduce disk load or use a faster disk
```
Comment (Ferroin) [inline health/health.d/dbengine.conf]: Phrasing suggestion for this:

'number of times in the last 10 minutes that the dbengine failed to completely flush data to disk'

Flows a bit better from the perspective of a native AmE speaker and more concretely explains the exact conditions that resulted in the alarm.

**Comment by Ferroin** [comment]
> I retest the PR with the last update, I could not raise the alarm again, but everything is working fine with it.

I ended up testing this one myself this time.  I was able to get the alarm to trigger by running in a VM with the storage directory being mounted over NFS with severe bandwidth limiting and intentionally using highly inefficient mount options for NFS itself.  It seems to me that this is only likely to trigger when you've got particularly slow storage and insane numbers of metrics (well into the hundreds of thousands, possibly more), though it's definitely good to have it.

**Pull request merged**